src/operator/nn/cast_storage-inl.h

```diff
@@ -0,0 +1,300 @@
+/*!
+ *  Copyright (c) 2017 by Contributors
+ * \file cast_storage-inl.h
```

Comment:
src/operator/nn is the folder where all MXNet ops will be moved to eventually. You can see utility functions for convolution and pooling have already been moved there.